[FX] Make module getattr wrapper proxy buffers

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73612

Previously, the `_module_getattr` method did not check for the presence of buffers. This made it so that code such as the code in the test case could not properly record operations on these buffer values. This PR makes `_module_getattr` look in `named_buffers` and better aligns it with `create_arg`:

https://github.com/pytorch/pytorch/blob/701fa16eed40c633d8eef6b4f04ab73a75c24749/torch/fx/_symbolic_trace.py#L239

Once this is landed the following can be deleted:

https://github.com/huggingface/transformers/blob/6ccfa2170cd0cb7438a89f801c9f39b6b939a1da/src/transformers/utils/fx.py#L431

Differential Revision: [D34568113](https://our.internmc.facebook.com/intern/diff/D34568113)

Commit:
[FX] Make module getattr wrapper proxy buffers

[ghstack-poisoned]

jamesr66a commented:
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34568113).

Commit:
Update on "[FX] Make module getattr wrapper proxy buffers"


Previously, the `_module_getattr` method did not check for the presence of buffers. This made it so that code such as the code in the test case could not properly record operations on these buffer values. This PR makes `_module_getattr` look in `named_buffers` and better aligns it with `create_arg`:

https://github.com/pytorch/pytorch/blob/701fa16eed40c633d8eef6b4f04ab73a75c24749/torch/fx/_symbolic_trace.py#L239

Differential Revision: [D34568113](https://our.internmc.facebook.com/intern/diff/D34568113)

[ghstack-poisoned]

zdevito commented on torch/fx/_symbolic_trace.py:
```diff
@@ -478,6 +478,12 @@ def _module_getattr(self, attr, attr_val, parameter_proxy_cache):
                         val_proxy = self.create_proxy('get_attr', n, (), {}, **kwargs)  # type: ignore[arg-type]
                         parameter_proxy_cache[n] = val_proxy
                     return parameter_proxy_cache[n]
+        if isinstance(attr_val, torch.Tensor):
```
maybe refactor this with the almost identical logic above. Potentially you want the param_shapes_constant and ParameterProxy logic to apply to buffers as well.

zdevito reviewed (approved):
Looks good, one potential cleanup, but I don't have enough context to know if if param_shapes_constant should apply to buffers as well.

Commit:
Update on "[FX] Make module getattr wrapper proxy buffers"


Previously, the `_module_getattr` method did not check for the presence of buffers. This made it so that code such as the code in the test case could not properly record operations on these buffer values. This PR makes `_module_getattr` look in `named_buffers` and better aligns it with `create_arg`:

https://github.com/pytorch/pytorch/blob/701fa16eed40c633d8eef6b4f04ab73a75c24749/torch/fx/_symbolic_trace.py#L239

Once this is landed the following can be deleted:

https://github.com/huggingface/transformers/blob/6ccfa2170cd0cb7438a89f801c9f39b6b939a1da/src/transformers/utils/fx.py#L431

Differential Revision: [D34568113](https://our.internmc.facebook.com/intern/diff/D34568113)

[ghstack-poisoned]

jamesr66a commented:
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34568113).

Commit:
Update on "[FX] Make module getattr wrapper proxy buffers"


Previously, the `_module_getattr` method did not check for the presence of buffers. This made it so that code such as the code in the test case could not properly record operations on these buffer values. This PR makes `_module_getattr` look in `named_buffers` and better aligns it with `create_arg`:

https://github.com/pytorch/pytorch/blob/701fa16eed40c633d8eef6b4f04ab73a75c24749/torch/fx/_symbolic_trace.py#L239

Once this is landed the following can be deleted:

https://github.com/huggingface/transformers/blob/6ccfa2170cd0cb7438a89f801c9f39b6b939a1da/src/transformers/utils/fx.py#L431

Differential Revision: [D34568113](https://our.internmc.facebook.com/intern/diff/D34568113)

[ghstack-poisoned]

Commit:
Update on "[FX] Make module getattr wrapper proxy buffers"


Previously, the `_module_getattr` method did not check for the presence of buffers. This made it so that code such as the code in the test case could not properly record operations on these buffer values. This PR makes `_module_getattr` look in `named_buffers` and better aligns it with `create_arg`:

https://github.com/pytorch/pytorch/blob/701fa16eed40c633d8eef6b4f04ab73a75c24749/torch/fx/_symbolic_trace.py#L239

Once this is landed the following can be deleted:

https://github.com/huggingface/transformers/blob/6ccfa2170cd0cb7438a89f801c9f39b6b939a1da/src/transformers/utils/fx.py#L431

Differential Revision: [D34568113](https://our.internmc.facebook.com/intern/diff/D34568113)

[ghstack-poisoned]

jamesr66a commented:
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34568113).

Commit:
Update on "[FX] Make module getattr wrapper proxy buffers"


Previously, the `_module_getattr` method did not check for the presence of buffers. This made it so that code such as the code in the test case could not properly record operations on these buffer values. This PR makes `_module_getattr` look in `named_buffers` and better aligns it with `create_arg`:

https://github.com/pytorch/pytorch/blob/701fa16eed40c633d8eef6b4f04ab73a75c24749/torch/fx/_symbolic_trace.py#L239

Once this is landed the following can be deleted:

https://github.com/huggingface/transformers/blob/6ccfa2170cd0cb7438a89f801c9f39b6b939a1da/src/transformers/utils/fx.py#L431

Differential Revision: [D34568113](https://our.internmc.facebook.com/intern/diff/D34568113)

[ghstack-poisoned]

jamesr66a commented:
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34568113).
